### PR TITLE
Clean phpdoc, use type hinting and ::class

### LINKED ISSUE #22: Add type hinting
The 2.0 version will support only php > 7 so we can use type hinting

### PR DESCRIPTION
**Pull request by notFloran**
Fix #22

### CODE REVIEW

File: src/Logger/Request/RequestLogger.php [hunk 1]
```diff
@@ -54,17 +59,10 @@ public function createContexts(Request $request)
             'request_locale' => $request->getLocale(),
             'request_auth_user' => $request->getUser(),
             'request_auth_has_password' => !is_null($request->getPassword()),
+            'request_encodings' => $request->getEncodings(),
+            'request_client_ips' => $request->getClientIps(),
         );
 
-        // Attributes from newer versions.
```
Comment (notFloran) [inline src/Logger/Request/RequestLogger.php:59]: newer versions = 2.4 and 2.5, we allow only SF3 so I removed the code

File: src/Provider/Session/SymfonySessionIdProvider.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class SymfonySessionIdProvider implements SessionIdProviderInterface
      * @param bool             $startSession
      * @param SessionInterface $session
      */
-    public function __construct($startSession = false, SessionInterface $session)
+    public function __construct(bool $startSession = false, SessionInterface $session)
```
Comment (nclsHart) [inline src/Provider/Session/SymfonySessionIdProvider.php:27]: This is not directly related to this PR, but arguments with default value should be at end of arguments list.

**Review by nclsHart — APPROVED** [review]

File: src/Provider/Session/SymfonySessionIdProvider.php (hunk 2 above)
Comment (notFloran) [reply src/Provider/Session/SymfonySessionIdProvider.php:27]: I create an issue #25
